iOS/MobileStarter/react-app/src/IModelDownloader.tsx

```diff
@@ -0,0 +1,115 @@
+    if (localBriefcases.length === 0) {
+      // This should never happen, since we just downloaded it, but check, just in case.
+      console.error("Error downloading iModel.");
+    }
+    return { minimalIModel: iModel, briefcase: localBriefcases[0] };
+  } catch (error) {
+    if (error instanceof BentleyError) {
+      if (error.errorNumber === IModelStatus.FileAlreadyExists) {
+        // When a download is canceled, the partial briefcase file does not get deleted, which causes
+        // any subsequent download attempt to fail with this error number. If that happens, delete the
+        // briefcase and try again.
+        try {
+          // When syncMode is SyncMode.PullOnly (which is what we use), briefcaseId is ALWAYS 0, so try
+          // to delete the existing file using that briefcaseId.
+          const filename = await NativeApp.getBriefcaseFileName({ iModelId: iModel.id, briefcaseId: 0 });
+          await NativeApp.deleteBriefcase(filename);
+          return downloadIModel(project, iModel, handleProgress);
+        } catch (_error) { }
+      } else if (error.errorNumber === BriefcaseStatus.DownloadCancelled && canceled) {
+        // When we call requestCancel, it causes the downloader to throw this error; ignore.
+        return { minimalIModel: iModel };
+      }
+    }
+    // There was an error downloading the iModel. Show the error
+    presentError("DownloadErrorFormat", error, "HubScreen");
+  }
+  return { minimalIModel: iModel };
+}
+
+export interface IModelDownloaderProps {
+  project: Project;
+  model: IModelInfo;
+  onDownloaded: (model: IModelInfo) => void;
+  onCanceled?: () => void;
+}
+
+export function IModelDownloader(props: IModelDownloaderProps) {
+  const { project, model, onDownloaded, onCanceled } = props;
+  const [progress, setProgress] = React.useState(0);
+  const [indeterminate, setIndeterminate] = React.useState(true);
+  const [downloading, setDownloading] = React.useState(false);
+  const [canceled, setCanceled] = React.useState(false);
+  const isMountedRef = useIsMountedRef();
+  const downloadingLabel = React.useMemo(() => i18n("HubScreen", "Downloading"), []);
+  const cancelLabel = React.useMemo(() => i18n("HubScreen", "Cancel"), []);
+
+  const handleProgress = React.useCallback((progressInfo: ProgressInfo) => {
+    if (isMountedRef.current) {
+      const percent: number = progressInfo.percent ?? (progressInfo.total ? Math.round(100.0 * progressInfo.loaded / progressInfo.total) : 0);
+      setProgress(percent);
+      setIndeterminate(false);
+    }
+    return isMountedRef.current && !canceled;
+  }, [canceled, isMountedRef]);
+
+  React.useEffect(() => {
+    if (downloading)
+      return;
+
+    const fetchIModel = async () => {
+      const newModel = await downloadIModel(project, model.minimalIModel, handleProgress);
+      if (!isMountedRef.current) return;
+      onDownloaded(newModel);
+    };
+    setDownloading(true);
+    fetchIModel();
```

Comment:
This should throw a linter warning (at least it does with the eslint setup we have on web).  Please add a disable comment.

Reply:
We don't have lint set up on the samples repository. @Carl-Hinkle do you want us to do that before 1.0?